gnomad/assessment/validity_checks.py

```diff
@@ -857,12 +863,53 @@ def vcf_field_check(
     return True
 
 
+def compare_related_global_and_row_lengths(
+    t: Union[hl.MatrixTable, hl.Table],
+    len_comp_globals_rows: Dict[str, List[str]] = None,
+) -> None:
+    """
+    Compare the lengths of global and row annotations.
+
+    :param t: Input MatrixTable or Table.
+    :param len_comp_globals_rows: Dictionary with row annotation (key) and list of associated global annotations (value) to compare.
+    :return: None
+    """
+    t = t.rows() if isinstance(t, hl.MatrixTable) else t
+    for row_field, global_fields in len_comp_globals_rows.items():
+        row_len = len(t.head(1)[row_field].collect()[0])
+        for global_field in global_fields:
+            global_len = hl.eval(hl.len(t[global_field]))
+            outcome = "Failed" if global_len != row_len else "Passed"
+            logger.info(
+                "%s global and row lengths comparison: Length of %s in"
+                " globals (%d) does %smatch length of %s in rows (%d)",
+                outcome,
+                global_field,
+                global_len,
+                "NOT " if outcome == "Failed" else "",
+                row_field,
+                row_len,
+            )
+
+
+def pprint_global_anns(t: Union[hl.MatrixTable, hl.Table]) -> None:
+    """
+    Pretty print global annotations.
+
+    :param t: Input MatrixTable or Table.
+    """
+    global_pprint = {g: hl.eval(t[g]) for g in t.globals}
+    pprint(global_pprint, sort_dicts=False)
+
+
 def validate_release_t(
     t: Union[hl.MatrixTable, hl.Table],
     subsets: List[str] = [""],
     pops: List[str] = POPS[CURRENT_MAJOR_RELEASE],
     missingness_threshold: float = 0.5,
-    monoallelic_expr: Optional[hl.expr.BooleanExpression] = None,
+    monoallelic_expr: Optional[
```

Comment:
Sorry @ch-kr , I committed suggestions but will update this and L257 above to whatever we decide. You suggested `unexpected_gt_expr` and I put forward `same_call_site_expr` but mine limits it to these two uses, mono and only het.

Reply:
Also my reasoning for not renaming originally was that it will be a breaking change but seems like a pretty minor one.

Reply:
I'm bad at naming things 😆 what about `gt_check_expr`? is that too general? otherwise `same_call_site_expr` works for me since we don't have another use for this field other than monoallelic or only het yet (though good point that renaming again in the future would be another breaking change)